examples/rvc-app/rvc-common/src/rvc-device.cpp

```diff
@@ -0,0 +1,107 @@
+#include "rvc-device.h"
+
+using namespace chip::app::Clusters;
+
+void RvcDevice::Init()
+{
+    mRunModeInstance.Init();
+    mCleanModeInstance.Init();
+    mOperationalStateInstance.Init();
+}
+
+void RvcDevice::HandleRvcRunChangeToMode(uint8_t NewMode, ModeBase::Commands::ChangeToModeResponse::Type & response)
+{
+    uint8_t currentMode = mRunModeInstance.GetCurrentMode();
+
+    switch (NewMode)
+    {
+    case RvcRunMode::ModeMapping: {
+        // change to mapping only allowed from the Idle state.
+        if (currentMode != RvcRunMode::ModeIdle)
+        {
+            response.status = to_underlying(ModeBase::StatusCode::kGenericFailure);
+            response.statusText.SetValue(chip::CharSpan::fromCharString("Change to the mapping mode is only allowed from idle"));
+            return;
+        }
+
+        mRunModeInstance.UpdateCurrentMode(NewMode);
+        // Set operational state to running
+        ChipError err =
+            mOperationalStateInstance.SetOperationalState(to_underlying(OperationalState::OperationalStateEnum::kRunning));
+        if (err != CHIP_NO_ERROR)
+        {
+            response.status = to_underlying(ModeBase::StatusCode::kGenericFailure);
+            response.statusText.SetValue(chip::CharSpan::fromCharString(err.AsString()));
```

Comment:
This is pointing the span to some random buffer that is not guaranteed to contain anything useful after the next time someone does something with a CHIP_ERROR...

This seems to highlight a problem in the API here: it only works when statusText can be pointed to static-lifetime strings (e.g. constant strings).

There are multiple instances of that in this file.

Reply:
The `SetValue` method is moving the value passed to it so that it retains a local copy. So I don't think there is an issue here.

Reply:
> The SetValue method is moving the value passed to it so that it retains a local copy.

No, it's not.  It's copying the Span, but it's not copying _the data the span points to_.

Reply:
I have removed these error checks since there is no way an error can be caused in this case. In makes for a simpler read as well. See [here](https://github.com/project-chip/connectedhomeip/pull/28719/commits).